### ISSUE TITLE
Github from-date

### ISSUE THREAD

**Issue by bhack** [chaoss/grimoirelab#428]
Are you currently regular testing github `from-date` in the config?
It seems that something is broken. I've tried just to set It to One or Two days before now and the log It Is full of api request to the github API.  It will not complete to crawl 1 day data in hours that instead It could be superfast.

**Comment by zhquan** [comment]
Hi @bhack 

It would be great if you could share the logs and the `setup.cfg`. To check for errors.

Best,
Quan

**Comment by bhack** [comment]
E.g.
```
[github]
api-token = <the yoken>
#enterprise-url = <YOUR_GITHUB_ENTERPRISE_URL>
raw_index = github_demo_raw
sleep-for-rate = true
sleep-time = "300"
from-date = 2021-05-04
enriched_index = github_demo_enriched
studies = [enrich_onion:github, 
           enrich_geolocation:user]
```
In the log:
```
 grimoire_elk.raw.elastic - INFO - [github] Incremental from: 2021-05-04 00:00:00+00:00 for https://github.com/tensorflow/tensorflow
.....
urllib3.connectionpool - DEBUG - https://api.github.com:443 "GET /repos/tensorflow/tensorflow/issues?state=all&per_page=100&direction=asc&sort=updated&since=2021-04-28T00%3A00%3A00%2B00%3A00 HTTP/1.1" 200 None
DEBUG - Archiving 3631ebd86d176efa0e2e4342c15fbda75857fd8d with https://api.github.com/repos/tensorflow/tensorflow/issues {'state': 'all', 'per_page': 100, 'direction': 'asc', 'sort': 'updated', 'since': '2021-04-28T00:00:00+00:00'}
```

Then I see that it is going to retrieve all the comments and all the reactions for every issue or pr that has also just a single update event since that datetime. Is this the logic? Cause I suppose it is going to saturate the API limits quite soon also with a medium traffic repository.

**Comment by zhquan** [comment]
Hi @bhack 

> Then I see that it is going to retrieve all the comments and all the reactions for every issue or pr that has also just a single update event since that datetime. Is this the logic?

No, Mordred has several categories to fetch different types of info. Take a look at these links:

- https://github.com/chaoss/grimoirelab-sirmordred/#github-
  - issue (by default)
  - pull request
  - repo
- https://github.com/chaoss/grimoirelab-sirmordred/#githubql-
- https://github.com/chaoss/grimoirelab-sirmordred/#github2-
  - issue
  - pull request

> Cause I suppose it is going to saturate the API limits quite soon also with a medium traffic repository.

Well, that depends on the number of the issues and prs. Any way you can add more tokens to increase the limit.

```
[github]
api-token = [TOKEN-1, TOKEN-2, ....]
```


Best,
Quan

**Comment by bhack** [comment]
So is it normal that it is going to retrieve all the comments and reactions with your default cfg? Do we need to limit [github] key with [github:issue] and [github:pr]?
https://github.com/chaoss/grimoirelab/blob/6881e50e0bb435883d447bbe39ee99b8bae6c0cf/default-grimoirelab-settings/setup.cfg#L65-L69

**Comment by zhquan** [comment]
> So is it normal that it is going to retrieve all the comments and reactions with your default cfg?

No, by default it will retrieve issues and pull requests, but not comments or reactions.

On https://chaoss.biterg.io/ you have some examples:
By default `github` set up you will have:
Issues:
- https://chaoss.biterg.io/app/kibana#/dashboard/GitHub-Issues
- https://chaoss.biterg.io/app/kibana#/dashboard/GitHub-Issues-Backlog
- https://chaoss.biterg.io/app/kibana#/dashboard/GitHub-Issues-Timing
- https://chaoss.biterg.io/app/kibana#/dashboard/c3da5c20-e1c3-11e8-8aac-ef7fd4d8cbad

Pull Requests:
- https://chaoss.biterg.io/app/kibana#/dashboard/GitHub-Pull-Requests
- https://chaoss.biterg.io/app/kibana#/dashboard/GitHub-Pull-Requests-Backlog
- https://chaoss.biterg.io/app/kibana#/dashboard/GitHub-Pull-Requests-Timing
- https://chaoss.biterg.io/app/kibana#/dashboard/9663d5a0-e1dc-11e8-8aac-ef7fd4d8cbad

But if you want to retrieve comments you have to activate [github2](https://github.com/chaoss/grimoirelab-sirmordred/#github2-)
- `github2:issue`: https://chaoss.biterg.io/app/kibana#/dashboard/4b6cdf80-33cd-11ea-b68b-31a1aa44b23a
- `github2:pull`: https://chaoss.biterg.io/app/kibana#/dashboard/bcb51ca0-3870-11ea-b68b-31a1aa44b23a

Best,
Quan

**Comment by bhack** [comment]
I don't want to retrieve comment but I see comments and reactions api request with the github section I've shared. Why?

**Comment by zhquan** [comment]
@bhack - sorry I misunderstood.

You are right, Perceval retrieves issues, comments, and reactions https://github.com/chaoss/grimoirelab-perceval/blob/master/perceval/backends/core/github.py#L306.

`github` and `github2` use the same backend in Perceval. On Mordred you have to choose what kind of data do you want to show.

Have a look the fields:
- `github`: https://github.com/chaoss/grimoirelab-elk/blob/master/schema/github_issues.csv
- `github2`: https://github.com/chaoss/grimoirelab-elk/blob/master/schema/github2_issues.csv

**Comment by bhack** [comment]
Thank you, 
honestly it is a little bit hard to clearly understand what is the minimal set of fields to minimize the github api requests and optimize API limits to populate the default dashboards that we see in the default ISSUE and PR menus dashboards. 
Can you distribute a good default?

Another point is: 
where we are going to set the restricted set of fields? Is it documented?

**Comment by zhquan** [comment]
> Can you distribute a good default?

The setup that you shared before is ok
```
#[github]
#api-token = <YOUR_API_TOKEN_WHERE>
#enterprise-url = <YOUR_GITHUB_ENTERPRISE_URL>
#raw_index = github_demo_raw
#sleep-for-rate = true
#sleep-time = "300"
#enriched_index = github_demo_enriched
```

To see the menu and the dashboards make sure the section `panels` is activated
```
[phases]
panels = true
```

> where we are going to set the restricted set of fields? Is it documented?

Sorry right now if you want to analyze `github` it will fetch [all these fields](https://github.com/chaoss/grimoirelab-elk/blob/master/schema/github_issues.csv). And here is the [github Perceval doc](https://perceval.readthedocs.io/en/latest/perceval.backends.core.html#module-perceval.backends.core.github)

**Comment by bhack** [comment]
> Sorry right now if you want to analyze github it will fetch all these fields. And here is the github Perceval doc
Mhhh.. it seems really a waste of the Github API resources/rate limitis for the standard Dashboard/Analysis when you need to populate the full project history on the first run other then imply to have the `no-archive = false` conf quite mandatory.

**Comment by vchrombie** [comment]
> Mhhh.. it seems really a waste of the Github API resources/rate limits for the standard Dashboard/Analysis when you need to populate the full project history on the first run other then imply to have the no-archive = false conf quite mandatory.

@bhack, thanks for the feedback.

Do you think we can close this issue as the query is resolved?

**Comment by bhack** [comment]
Where we can track this Feature request?

**Comment by vchrombie** [comment]
> Where we can track this Feature request?

It would be great if you can open a new issue since the idea is different from this issue. WDYT?

**Comment by bhack** [comment]
The idea is the same. I just supposed that it was a bug but instead it is a feature request.

**Comment by vchrombie** [comment]
> The idea is the same. I just supposed that it was a bug but instead it is a feature request.

Yup, I understand it. It would be great if you open a new issue for more discussion on this feature request. :slightly_smiling_face: 
Let me know if you need any help.

**Comment by bhack** [comment]
In what repository?

**Comment by vchrombie** [comment]
You can open it in this repository itself.
Thanks.

**Comment by bhack** [comment]
I don't understand why I need to open a new one for the same repo but ok. It is https://github.com/chaoss/grimoirelab/issues/432

**Comment by jgbarah** [comment]
A bit of context here, since I was a part of the original decisions that made this work this way.

The original GrimoreLab assumption was: "we want to get all the data for a repo, store it into a database, and then access the database to get specific results". This is probably optimum for analyzing the full story of repositories, but i agree is different from your use case, where (if i understand well), you only need issues and pull requests since some time ago (some days ago, for example).

I think this is way @vchrombie suggested to open a feature request, because the current functioning is intended, with the original assumptions of GrimoireLab.

Now, several years later, there are many different use cases, and some of them don't fit these original assumptions. I'm afraid your case, @bhack, is one of them. It requires a re-thinking of the basic assumptions, which is quite interesting, but at the same time, more difficult than just fixing a bug.

This said, I'd also like GrimoireLab having into account your use case: it is going mainly a matter of finding resources to allocate to that, I think. Thanks in any case for the discussion.

**Comment by bhack** [comment]
@jgbarah Have you tried to import the full history with all the comments etc.(as now) with a very large project with high traffic. I suppose that with the API rate limits it will requires days or weeks or not?

**Comment by sduenas** [comment]
> @jgbarah Have you tried to import the full history with all the comments etc.(as now) with a very large project with high traffic. I suppose that with the API rate limits it will requires days or weeks or not?

Yes, it might take some days. Jesús explained the idea behind this but also he missed to explain that we do that to not go twice. Sometimes you want to analyze issues and later you realized that you want the comments, so you need to go again to the repo and waste extra time. Of course, if this is not your use case, it's a problem because you want a fast analysis.

In any case, thank you for opening the discussion. We can close this issue, because your question is already resolved, and keep discussing about it on #432.
